contrib/synthetic_datagen/synthetic_datagen.py

```diff
@@ -0,0 +1,164 @@
+
+    # Types of distributions
+    # Equal     - No change in lengths
+    # Uniform   - Uniform distribution across the min_max intervals
+    # Normal    - Standard normal distribution centered over the mean
+    if distribution == "uniform":
+        return (
+            random_generator.uniform(low=other_args["input_min"], high=other_args["input_max"], size=num_samples),
+            random_generator.uniform(low=other_args["output_min"], high=other_args["output_max"], size=num_samples),
+        )
+    elif distribution == "normal":
+        return (
+            random_generator.normal(loc=other_args["input_mean"], scale=other_args["input_sd"], size=num_samples),
+            random_generator.normal(loc=other_args["output_mean"], scale=other_args["output_sd"], size=num_samples),
+        )
+    elif distribution == "equal":
+        # Using the library for consistency
+        return (
+            random_generator.normal(loc=other_args["input_len"], scale=0, size=num_samples),
+            random_generator.normal(loc=other_args["output_len"], scale=0, size=num_samples),
+        )
+    else:
+        raise RuntimeError("Unknown distribution requested : " + str(args.distribution))
+
+def make_one_sample(vocab : list, tokenizer, req_sample_size : int):
+    tokens = random.sample(vocab, req_sample_size)
+    Logger.debug(f"Generated sample tokens : {tokens}")
+    return tokenizer.decode(tokens, skip_special_tokens = False)
+
+def make_dataset(args):
+
+    model = args['model']
+    num_samples = args['num_samples']
+
+    dataset_info = {
+        'model': model,
+        'num_samples': num_samples,
+        'distribution': args['distribution']
+    }
+
+    # vLLM will add special tokens when running inference. 
+    # This can introduce a minor variation in the vLLM reported token count
+    tokenizer = Tokenizer.from_pretrained(
+        model
+        )
+    
+    # Use token ids instead of actual tokens
+    vocab = list(range(0, tokenizer.get_vocab_size()))
+
+    input_lengths, output_lengths = gen_io_lengths(
+        num_samples=num_samples, 
+        distribution=dataset_info['distribution'],
+        other_args=args
+        )
+
+    # dtype conversion due to output type of random sampling
+    input_lengths, output_lengths = input_lengths.astype(dtype=int).tolist(), output_lengths.astype(dtype=int).tolist()
+    Logger.debug(f"Input and Output lengths : {list(zip(input_lengths, output_lengths))}")
+    
+    dataset_info["io_lengths"] = list(zip(input_lengths, output_lengths))
+
+    dict_items = []
+    for si, (input_len, output_len) in enumerate(zip(input_lengths, output_lengths)):
+        sample = make_one_sample(vocab, tokenizer, input_len)
+        Logger.debug(f"Sample : {sample}")
+        dict_items.append({
+            "index": "custom-"+model+"-data-" + str(si),
+            "question": sample,
+            "tok_input_length": input_len,
+            "tok_output_length": output_len,
```

Comment:
```suggestion
            "tok_output_length": output_len,
            "system_prompt": "",
```

`system_prompt` must be set. See #85.